gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -317,10 +324,61 @@ MacroExpander::expand_crate ()
   // expand module tree recursively
 
   // post-process
+  feature_gate ();
 
   // extract exported macros?
 }
 
+void
+MacroExpander::feature_gate ()
```

Comment:
So I would move this function to the SessionManager as well. It's good, but not really related to macro expansion. It simply happens *after* macro expansion in the pipeline.